### PR TITLE
Upgrade to discord.js v14

### LINKED ISSUE #14: Upgrade to discord.js v14
Package needs upgrading to discord.js v14

### PR DESCRIPTION
**Pull request by olivier-b-be**
Resolves #14 

Breaking changes with upgrading discord.js v13 to v14

## Changes
- Classes with private/protected constructors have been changed to use `Reflect.construct()` method to be able to create instances of the classes
- Updated certain magic number values to use the new enum types eg. `GuildVerificationLevel`, `ChannelType`, etc.
- Removed `getChannel()` in `BaseMocks`, constructor is not available anymore so no base generic channel to be used
- Updated default properties
- Removed the `.only` set in `CustomMocksTest`

### CODE REVIEW

**Comment by olivier-b-be** [comment]
Have at me :D

I upgraded the version in package.json and package-lock.json, I assume that's enough for a new release but not entirely sure.

File: package.json [hunk 1]
```diff
@@ -48,9 +48,12 @@
...
     "sinon": "^9.2.1",
     "ts-mocha": "^7.0.0",
-    "typescript": "^4.0.3"
+    "typescript": "^4.9.5"
   },
   "dependencies": {
-    "discord.js": "^13.0.1"
+    "discord.js": "^14.7.1"
+  },
+  "overrides": {
```
Comment (LamboCreeper) [inline package.json]: Why is this required?

**Review by LamboCreeper — COMMENTED** [review]
Looks good, thanks! Just one question.

File: package.json (hunk 1 above)
Comment (olivier-b-be) [reply package.json]: There was an issue with the discord-api-types for v14, it was something mentioned in the Discord.JS discord FAQ channel but it's possible that they fixed it with the 14.8.0 release. I'll update the package and see whether I get the problem or not (was an issue with the guild channel types not being correct for sending a message in them)

**Review by LamboCreeper — APPROVED** [review]
